### ISSUE TITLE
[Featured Blog] Card link generates overly verbose user experience

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#452]
## Issue summary

The featured blog card links wrap the entirety of the card content:

1. Image `alt` text
2. Post title
3. Post date and author
4. Post excerpt
5. …and other content

This generates an overly verbose user experience for screen reader and voice dictation users. Screen readers announce all content at once, or when navigating via virtual cursor, announce "link" along with each piece of content.

> "Welcome Home, Hem founder Petrus "Palmér" Welcome Home, Hem founder Petrus "Palmér" MARCH 2018SCOTT VINKLE Hem means home, and home is where the heart is. For Hem founder Petrus Palmér and his family, home is located in a remodeled blacksmith’s workshop situated in the Stockholm... Read more: Welcome Home, Hem founder Petrus &quot;Palmér&quot; 10 comments"

Voice dictation users may not be able to call out the command to click the link with the post title; this is not the first piece of content which makes up the computed accessible name (the first piece is the image `alt` as this is first in the DOM within the link.)

<details>
  <summary>Screenshots</summary>

<img width="1160" alt="Dawn theme, featured blog section cards are highlighted." src="https://user-images.githubusercontent.com/1392632/130131372-6bec43be-ea3d-4a8f-bc10-71f7196947f0.png">

</details>

#### HTML

```html
<article class="article-card" aria-labelledby="Article-10759274554">
  <a
    href="/blogs/news/welcome-home-hem-founder-petrus-palmer"
    class="article-content motion-reduce"
  >
    <div class="article-card__image-wrapper">
      <div class="article-card__image media media--landscape">
        <!-- Image… -->
      </div>
    </div>
    <div class="article-card__info">
      <header class="article-card__header">
        <h2 class="article-card__title" id="Article-10759274554">
          Welcome Home, Hem founder Petrus "Palmér"
        </h2>
        <!-- Meta content… -->
      </header>
      <p class="article-card__excerpt rte-width">
        <!-- Excerpt… -->
      </p>
      <div class="article-card__footer">
        <span
          class="link article-card__link"
          aria-label="Read more: Welcome Home, Hem founder Petrus &amp;quot;Palmér&amp;quot;"
        >
          Read more
        </span>
        <span>10 comments</span>
      </div>
    </div>
  </a>
</article>
```

## Steps to reproduce

1. Go to a Dawn theme store with any supported browser
2. With an active screen reader, <kbd>Tab</kbd> to or use the virtual cursor to navigate the featured blog listing

### Expected

- <kbd>Tab</kbd> to announce the post title only, virtual cursor to consume other card content with intended semantic meaning

### Actual

- <kbd>Tab</kbd> announces all content within the link, virtual cursor announces all content items with "link" semantics

### HTML structure…

1. Move the link element to only wrap the post title text.
2. Add a appropriate heading element by swapping the `h2.article-card__title` element with an `h3`.
3. Swap the `span.link` with an `a` element, setting its `href` as the link from the title.

### CSS modifications

1. Use CSS to wrap/cover the card with the link element to re-apply the large click/touch target.

Review [Inclusive Components – Card](https://inclusive-components.design/cards/) for more the rationale behind these changes.

#### HTML

```html
<article class="article-card" aria-labelledby="Article-10759274554">
  <div class="article-card__image-wrapper">
    <div class="article-card__image media media--landscape">
      <!-- Image… -->
    </div>
  </div>
  <div class="article-card__info">
    <header class="article-card__header">
      <h3 class="article-card__title" id="Article-10759274554">
        <a
          href="/blogs/news/welcome-home-hem-founder-petrus-palmer"
          class="article-content motion-reduce"
        >
          Welcome Home, Hem founder Petrus "Palmér"
        </a>
      </h3>
      <!-- Meta content… -->
    </header>
    <p class="article-card__excerpt rte-width">
      <!-- Excerpt… -->
    </p>
    <div class="article-card__footer">
      <a
        href="/blogs/news/welcome-home-hem-founder-petrus-palmer"
        class="link article-card__link"
        aria-label="Read more: Welcome Home, Hem founder Petrus &amp;quot;Palmér&amp;quot;"
      >
        Read more
      </a>
      <span>10 comments</span>
    </div>
  </div>
</article>
```

#### CSS

```css
.article-card {
  position: relative;
  /* … */
}

a.article-content.motion-reduce:after {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **WCAG Principal:** Perceivable, Robust
- **WCAG SC:** [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html), [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)
- **Severity:** High
- **Effort:** High
- **Related:** https://github.com/Shopify/dawn/issues/692

**Comment by svinkle** [comment]
Closed via https://github.com/Shopify/dawn/issues/692#issuecomment-1071086899.
